lib/cluster/node.js

```diff
@@ -35,6 +38,85 @@ const ClusterState = require('./state');
 const ClusterCommand = require('./command');
 const kuzzleStateEnum = require('../kuzzle/kuzzleStateEnum');
 
+
+/**
+ * Test an IP address and determine if it's in the public or private range.
+ *
+ * @param  {String}  ip
+ * @return {Boolean}
+ */
+function isPrivateIP(ip) {
+  assert(net.isIP(ip) !== 0, `[CLUSTER] "${ip}" is not a valid IP address`);
+
+  if (net.isIPv6(ip)) {
+    const prefix = ip.split(':')[0];
+
+    return (prefix.startsWith('fd') && prefix.length === 4) || prefix === 'fe80';
+  }
+
+  // IPv4
+  const exploded = ip.split('.').map(s => Number.parseInt(s));
+
+  return exploded[0] === 10
+    || (exploded[0] === 172 && exploded[1] >= 16 && exploded[1] <= 31)
+    || (exploded[0] === 192 && exploded[1] === 168)
+    // Should never be considered: this range is reserved for internal uses only
+    || exploded[0] === 127
+    // 169.254.x.x addresses are... particular. I *think* they are usable if
+    // multiple machines share the same DHCP rules, but I don't really know.
+    // This might be an edge case.
+    // For now I'll consider them as private addresses, but we might want to
+    // consider them as internal ones if that poses problems in the future.
+    || (exploded[0] === 169 && exploded[1] === 254);
```

Comment:
I was thinking about APIPA addresses this weekend. If I remember correctly they do not allow network communication with other addresses of the same type. They are more of a stopgap to allow more "classical" services that need an IP address to start without any problems. To me we should return an error if the only available addresses are APIPAs. Otherwise we may find ourselves in error cases that are very annoying to debug.

What do you think?

Reply:
I thought I read something like that, but I was not entirely sure.

If APIPA addresses are not reachable from other machines, then we must consider them as internal addresses, exactly like loopback ones.

I'll amend my PR to reflect that.